tests/end2end/features/test_fileselect_bdd.py

```diff
@@ -0,0 +1,62 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016-2020 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+
+import pytest_bdd as bdd
+bdd.scenarios('fileselect.feature')
+
+
+@bdd.when(bdd.parsers.parse('I set up a fake single file fileselector '
+                            'selecting "{choosefiles}"'))
+def set_up_single_fileselector(quteproc, py_proc, choosefiles):
+    """Set up fileselect.single_file.command to select the file."""
+    set_up_fileselector(
+        quteproc=quteproc,
+        py_proc=py_proc,
+        choosefiles=choosefiles,
+        setting_cmd='fileselect.single_file.command'
```

Comment:
I'm a bit confused about those names (both here and below). I'd prefer something like:

- `selected_files` or perhaps even just `files` (rather than `choosefiles`)
- `setting` or `setting_name` (it's not a command)

Reply:
Fixed